doc/sections/module.md

```diff
@@ -1,45 +1,84 @@
-module is accessed using a full qualification `lib.boolean`. To illustrate the
-behaviour of the system, consider the following example.
-```bash
-working_directory  # lambdapi is invoked in that directory
-├── main.lp        # this module must always be refered as "main"
-├── bool.lp        # this module must always be refered as "bool"
-├── church
-│   ├── nat.lp     # this module must always be refered as "church.nat"
-│   └── list.lp    # this module must always be refered as "church.list"
-└── scott
-    ├── nat.lp     # this module must always be refered as "scott.nat"
-    └── list.lp    # this module must always be refered as "scott.list"
+Lambdapi has a (very) light module system based on file paths. It allows us to
+split our developments into several files, and even several folders.  The main
+thing to know is that a file holds a single module,  and accessing this module
+in other files requires using its *full* module path.
+
+#### Module path
+
+The module path that corresponds to a source file is defined using the name of
+the file, but also the position of the file under the *library root* (a folder
+under which the standard library and all packages are installed).
+
+The typical case is when we want to access a module of some installed library,
+for example the standard library. In that case, the module path is built using
+the file path as follows: if our source file is at `<LIB_ROOT>/std/bool.lp` it
+is accessed with module path `std.bool`.  And if there are nested folders then
+the module path gets more members. File `<LIB_ROOT>/std/a/b/c/d.lp` has module
+path `std.a.b.c.d`.
+
+By default, every modules are looked up under the library root. However, there
+are cases where the files we want to work with are not (yet!) placed under the
+library root.  The typical case is when a library (or package) is still  under 
+development. In that case,  what we can do is map the development folder under
+the library root, similarly to what would happen when mounting a volume in our
+file system. There are two ways of doing that, the first one being to used the
+`--map-dir MOD:DIR` command line option.  However, the best way is to use a
+package configuration file.
+
+#### Package configuration file
+
+A package configuration file can be placed at the root of the source tree of a
+library (or package) under development. It should be named `lambdapi.pkg`, and
+contain the following fields (an example is given below for the syntax):
+ - `package_name` giving a globally unique name for the package being defined.
+   This is not yet used,  but will be necessary when we eventually publish our
+   packages online so that they can be automatically downloaded when our users
+   need them. (The idea is to come up with a system similar to Cargo in Rust.)
+ - `root_path` gives the module path under which the library is to be  placed.
+   Assuming that our configuration file is at `<REPO_ROOT>/lambdapi.pkg`, this
```

Comment:
can the configuration file be somewhere else? what's the semantics in this case?

Reply:
I don't understand your question, what are you referring to?

Reply:
I find the beginning of the sentence confusing. Could you please write:

A package configuration file  `<REPO_ROOT>/lambdapi.pkg` can be placed at the root of the source tree of a
library (or package) under development. It should contain the following fields (an example is given below for the syntax): ...

Reply:
Your version is not precise enough: the root of a package might not coincide with the root of a repository. For instance, it may be that several related but independent packages are developed in the same repository.